### PR TITLE
Extend API documentation

### PR DESCRIPTION
**Pull request by antoniotorresm**
This includes:

* Section about command/param info in usage guide
* Section about metadata retrieval in usage guide
* Guide about differences between CLI and API
* Access control guide (management of roles, privileges and permissions).
* Guide about API contexts
* JSON-RPC usage guide and JSON-to-Python conversion
* Notes about types in API Reference

### CODE REVIEW

**Comment by antoniotorresm** [comment]
Linter is failing because of changes in API Reference for type pages. This can be waived since these notes were added manually as they are not automated (most of them are derived from source comments).

File: doc/api/jsonrpc_usage.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+```
+
+## Converting JSON-RPC requests to Python
+
+FreeIPA provides methods to convert JSON-RPC requests to Python format.
+
+```python
+from ipaserver.rpcserver import jsonserver
+
+json_request = '{"method":"user_find","params":[[""],{}],"id":0}'
```
Comment (abbra) [inline doc/api/jsonrpc_usage.md:121]: This needs `freeipa-python-ipaserver` package (on Fedora). I wonder if we can refactor `ipaserver.rpcserver` and move `marshal()`/`unmarshal()` code into `ipalib.rpc` and call it from there? `unmarshal()` does not depend on `self` at all, `marshal()` only needs a `self.api.env.debug` one that can be passed as a debug argument to that function...

Comment (antoniotorresm) [reply doc/api/jsonrpc_usage.md:122]: Added line about the `freeipa-python-ipaserver` package. I've noted the `rpcserver` refactor for a future PR.

**Comment by antoniotorresm** [comment]
The API doc validation will actually need a different PR in order to pass, since the comments this PR adds should be under the "manual notes" section (non-automated content).

**Comment by antoniotorresm** [comment]
We need to merge https://github.com/freeipa/freeipa/pull/6738 and update this PR accordingly before merging it.

**Comment by antoniotorresm** [comment]
PR was rebased to account for changes in #6738, doc validation now succeeds. Awaiting additional comments / ack.

**Comment by rcritten** [comment]
LGTM. @abbra do you have any further comments?

**Comment by abbra** [comment]
No, I have no more comments. Thanks a lot! Let's merge this!

**Comment by flo-renaud** [comment]
master:

* 88d8534e49362afb73512b9b97447711169f4df4 Extend API documentation

**Pull request closed without merging**